api/api.proto

```diff
@@ -1203,6 +1203,143 @@ service OpenStorageRole {
       };
     }
 }
+
+// ## OpenStorageFilesystemTrim Service
+// This service provides methods to manage filesystem trim operation on a
+// volume. 
+//
+// This operation runs in the background on a **mounted volume**. If the volumes
+// are not mounted, these API return error.
+//
+// Once the filesystem trim operation is started, the clients have to poll for
+// the status of the background operation using the
+// `OpenStorageFilesystemTrim.GetStatus()` rpc request
+//
+// A typical workflow involving filesystem trim would be as follows
+// 1. Attach the volume
+//    `OpenStorageMountAttachClient.Attach()`
+// 2. Mount the volume
+//    `OpenStorageMountAttachClient.Mount()`
+// 3. Start the filesystem trim operation by issuing a grpc call to
+//    `OpenStorageFilesystemTrimClient.Start()`
+//    This call returns immediately with a status code indicating if the
+//    operation was successfully started or not.
+// 4. To get the status of the Filesystem Trim operation, issue a grpc call to 
+//    `OpenStorageFilesystemTrimClient.GetStatus()`
+// 5. To stop the Filesystem Trim operation, issue a grpc call to
+//    `OpenStorageFilesystemTrimClient.Stop()`
+service OpenStorageFilesystemTrim {
+
+  // Start a filesystem Trim background operation on a mounted volume
+  rpc Start(SdkFilesystemTrimStartRequest)
+    returns (SdkFilesystemTrimStartResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/start"
+        body: "*"
+      };
+    }
+
+  // Get Status of a filesystem Trim background operation on a mounted
+  // volume, if any
+  rpc GetStatus(SdkFilesystemTrimGetStatusRequest)
+    returns (SdkFilesystemTrimGetStatusResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/get-status"
+        body: "*"
+      };
+    }
+
+  // Stop a filesystem Trim background operation on a mounted volume, if any
+  rpc Stop(SdkFilesystemTrimStopRequest)
+    returns (SdkFilesystemTrimStopResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/stop"
+        body: "*"
+      };
+    }
+  }
+
+// ## OpenStorageFilesystemCheckService
+// This service provides methods to manage filesystem check operation on a
+// volume. 
+//
+// This operation is run in the background on an **unmounted volume**.
+// If the volume is mounted, then these APIs return error.
+//
+// Once the filesystem check operation(either CheckHealth() or FixAll()) is
+// started, the clients have to poll for the status of the background operation
+// using the `OpenStorageFilesystemcheck.CheckHealthGetStatus()` rpc request or
+// `OpenStorageFilesystemCheck.FixAllGetStatus()` rpc request.
+//
+// **Note: CheckHealth() and FixAll() cannot run in parallel for the same volume**
+
```

Comment:
need a `//` here to show that this is still part of the same documentation. If not, the documentation will start with line 1276. We can fix this after. Please feel free to merge.